src/commands/new_sale_ticket.rs

```diff
@@ -0,0 +1,49 @@
+use candid::Encode;
+use clap::Parser;
+
+use crate::{
+    commands::transfer::HexSubaccount,
+    lib::{
+        signing::{sign_ingress_with_request_status_query, IngressWithRequestId},
+        AnyhowResult, TargetCanister,
+    },
+    SnsCanisterIds,
+};
+
+/// Attempts to create a new Ticket for the caller.
+#[derive(Parser)]
+pub struct NewSaleTicketOpts {
+    // TODO: check this description
```

Comment:
Is this still a TODO? Can you include the sdk ticket that will solve this if its dependent on the implementation work done by Mario?